src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -160,7 +160,7 @@ private bool TryGetHostProject(ProjectId projectId, out AbstractProject project)
             }
 
             // first make sure we can edit the document we will be updating (check them out from source control, etc)
-            var changedDocs = projectChanges.SelectMany(pd => pd.GetChangedDocuments()).ToList();
+            var changedDocs = projectChanges.SelectMany(pd => pd.GetTextChangedDocuments()).ToList();
             if (changedDocs.Count > 0)
             {
                 this.EnsureEditableDocuments(changedDocs);
```

Comment:
I think EnsureEditableDocuments should still filter out null filepath in here

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,1117

if we want to see when such thing can happens, we can check null path and add NFW explicitly there.

Reply:
I think you should do this though.

Reply:
It is unclear what EnsureEditableDocuments should do when GetFilePath() returns null. I filed a bug https://github.com/dotnet/roslyn/issues/26376